app/deliver_grant_funding/routes/reports.py

```diff
@@ -348,6 +351,76 @@ def list_group_questions(grant_id: UUID, group_id: UUID) -> ResponseReturnValue:
     )
 
 
+@deliver_grant_funding_blueprint.route(
+    "/grant/<uuid:grant_id>/task/<uuid:form_id>/groups/add",
+    methods=["GET", "POST"],
+)
+@has_grant_role(RoleEnum.ADMIN)
+@auto_commit_after_request
+def add_question_group_name(grant_id: UUID, form_id: UUID) -> ResponseReturnValue:
+    form = get_form_by_id(form_id)
+    group_name = request.args.get("name", None)
+
+    parent_id = request.args.get("parent_id", None)
+    parent = get_group_by_id(UUID(parent_id)) if parent_id else None
+
+    wt_form = GroupForm(name=group_name)
+
+    if wt_form.validate_on_submit():
+        # todo: check the name would be valid? - I think we have some precedent for this in the i.e add new grant flow
+        return redirect(
+            url_for(
+                "deliver_grant_funding.add_question_group_display_options",
+                grant_id=grant_id,
+                form_id=form_id,
+                name=wt_form.name.data,
+                parent_id=parent.id if parent else None,
+            )
+        )
+
+    return render_template(
+        "deliver_grant_funding/reports/add_question_group_name.html",
+        grant=form.section.collection.grant,
+        db_form=form,
+        form=wt_form,
+        parent=parent,
+    )
+
+
+@deliver_grant_funding_blueprint.route(
+    "/grant/<uuid:grant_id>/task/<uuid:form_id>/groups/add/display_options",
+    methods=["GET", "POST"],
+)
+@has_grant_role(RoleEnum.ADMIN)
+@auto_commit_after_request
+def add_question_group_display_options(grant_id: UUID, form_id: UUID) -> ResponseReturnValue:
+    form = get_form_by_id(form_id)
+    group_name = request.args.get("name", None)
+
+    if not group_name:
+        return abort(400, "A group name must be provided to add a question group")
```

Comment:
nit: maybe not sure about using query params for this, in part because:
1. url encoded arbitrary text loosk gross
2. as a principle, do we want use session+pydantic to serialize data for multi-page flows?